### ISSUE TITLE
current develop branch has broken sprite batch rendering

### ISSUE THREAD

**Issue by GrafSeismo** [MonoGame/MonoGame#4903]
Version 3.6.0.447 does not render anything between begin and end anymore, must move back to stable 3.5.x

**Comment by mrhelmut** [comment]
Which platform / type of project?

**Comment by GrafSeismo** [comment]
Win10 Pro, self made game.

Btw. 3.5. do have a lot more issues :c

**Comment by cra0zy** [comment]
What project are you using DesktopGL, Windows Direct X or UWP?

**Comment by tomspilman** [comment]
Most likely this is in one of the GL platforms as the unit tests just run against the Windows DirectX build.

**Comment by GrafSeismo** [comment]
I guess its DirectX under UAP/UWP, the strange part is that i do not get any error message just a blanck screen in the color i clear it out. No Font, no texture, nothing.

**Comment by tomspilman** [comment]
I'll give this a test today and report back.

**Comment by tomspilman** [comment]
I just did some rendering with `SpriteBatch` under UWP and it seems to be working fine.

Can you share a specific simple code snippet I can test?  What version of Windows 10 are you running?

**Comment by GrafSeismo** [comment]
Thank you very much for investigating! I am using a Microsoft Windows 10 Pro installation with version 10.0.10586 as i select this as minimum version in projects properties.
I found an issue with my display manager that get the list of available sceen sizes now in opposit order than XNA deliveres it. I fixed that without any progress in drawing. So i could not find a reason for not draw anything yet. I will try to start a new project and copy code until it stops drawing as long as i find the part that is reponsible for no drawing anything at all. I could draw with an earlier 3.6 release but had a lot issues with not fully implemented XACT Audio. So i will try to get more details for you at weekend. So i will enhace information soon.

Some Pre settings:
IsMouseVisible = false;
IsFixedTimeStep = true;
PreferredBackBufferFormat = Color;
PreferredDepthStencilFormat = Depth16;

// Windows size is not editable

// in general 

        protected override void Draw(GameTime gameTime)
        {
            if (IsActive == false)
            {
                return;
            }
            // It is Deffered
            Render.SpriteBatch.Begin(Render.SpriteSortMode);
            // Here i draw to different times different to no components.
            base.Draw(gameTime);
            // this screen manager draws similar as the xna sample for game state management
            Screen.Draw();
            // until console is not active it get skipped (but do not draw also)
            Console.Draw();
            // This is for mouse cursor
            Input.Draw();
            // End of main draw circle
            Render.SpriteBatch.End();
        }
    }

**Comment by tomspilman** [comment]
> I found an issue with my display manager that get the list of available sceen sizes now in opposit order than XNA deliveres it.

Note this changed recently to match XNA behavior... see https://github.com/mono/MonoGame/pull/4879.

> I could draw with an earlier 3.6 release but had a lot issues with not fully implemented XACT Audio.

XAct support will be getting better, but we're still a long way away from full compatibility with XNA.

**Comment by GrafSeismo** [comment]
Well dellis1972 said, XNA **seems** to report the DisplayModes in size order from small to large... I want to bespeak that i have an engine using that feature since years and had to change the order after you changed the order from small to large. XNA delivered it in opposit order because first element is most important due to the fact it is the maximum possible/usefull screen size. But that is not a major issue and anyone can change the order very easy. So from my perpective i can handle it ;)

**Comment by GrafSeismo** [comment]
Ok, please give this a try. I started a new project from template for a new MG windows universal project and added some rendering of a texture. Than drop the reference to the nuget package for MonoGame. Than add the MonoGame MonoGame.Framework.WindowsUniversal.sln project (you should have a copy of current repository) to your new game project and reference the MonoGame.Framework.WindowsUniversal (not the MonoGame.Framework.net.WindowsUniversal) to the game. En voila, you do not see any rendering any more.

**Comment by tomspilman** [comment]
> XNA delivered it in opposit order because first element is most important due to the fact it is the maximum possible/usefull screen size. 

Are you saying XNA returns the display modes in large to small order?  @dellis1972 ?


> Ok, please give this a try.

I'll try it out later today!

**Comment by dellis1972** [comment]
weird.. will need to double check... I really hope its not driver dependent that would really be a pain :(

**Comment by GrafSeismo** [comment]
I try to fork MG for a change i have made for delivering a proper desktop screen size in UAP in GraphicsAdapter
```
#elif WINDOWS_UAP
                var visibleBounds = ApplicationView.GetForCurrentView().VisibleBounds;
                var rawPixelsPerViewPixel = DisplayInformation.GetForCurrentView().RawPixelsPerViewPixel;
                return new DisplayMode((int)(visibleBounds.Width  * rawPixelsPerViewPixel + 0.5f),
                                       (int)(visibleBounds.Height * rawPixelsPerViewPixel + 0.5f),
                                       SurfaceFormat.Color);
#else
                return new DisplayMode(800, 600, SurfaceFormat.Color);
#endif
```

**Comment by theZMan** [comment]
The word from 'someone who would know' is that you should not assume any sort order on this property. The order is platform/implementation/who-knows-what-other-factors dependent.

So if you require a sort order, sort it yourself.

We should probably update our docs to reflect this.

**Comment by GrafSeismo** [comment]
Well, Microsoft did not document the order, so i am fine with any order, but my engine depends on the order and needed it as it was (on windows platform). I tested it very accurate due to specific comfortable features i wanted to deliver, but now it is useless due to missing support by UAP window. therefor any documetation about the order seems to be the best solution. May you can document it for all platforms.

**Comment by cra0zy** [comment]
The documentation will be, the displays aren't sorted in any order... we aren't gonna be documenting this per platform. If you want them sorted, use Linq, you literally need 1 line of code to sort them :|

**Comment by dellis1972** [comment]
We added a sort order in #4879
<https://github.com/mono/MonoGame/pull/4879> that
will be applied to all the platforms.

**Comment by GrafSeismo** [comment]
@cra0zy sorry if i made you angry, i am not a native speaker for english language. My only point was that i had measured XNA had an order that started with the highest possible screen size. So please ignore my intentions or wiches, it is up to you only what this project document or implement.

**Comment by cra0zy** [comment]
> We added a sort order in #4879

Good.

> @cra0zy sorry if i made you angry, i am not a native speaker for english language. My only point was that i had measured XNA had an order that started with the highest possible screen size. So please ignore my intentions or wiches, it is up to you only what this project document or implement.

Oh, sorry if I gave you that impression, I was not angry :/ I guess I need to work a bit on how I talk in English (English is not my mother tongue eithger).

**Comment by GrafSeismo** [comment]
Ok i uploaded a new pull request for the GraphicsAdapter to return the current screen size for WINDOWS_UAP.

**Comment by GrafSeismo** [comment]
Second test right now with current development SDK 3.6.0.525 failed also.
Start a new Windows 10 Universal project from template.
Open Content.mgcb and add a Texture.png. I took the Square44x44Logo.scale-200.png from Assets.
(Btw. the upcomming window show un-readable text, hidden by its buttons!)
Change game class as followed and you will not see any rendering...
```
using Microsoft.Xna.Framework;
using Microsoft.Xna.Framework.Graphics;
using Microsoft.Xna.Framework.Input;

namespace Game2
{
    /// <summary>
    /// This is the main type for your game.
    /// </summary>
    public class Game1 : Game
    {
        private readonly GraphicsDeviceManager graphics;
        private SpriteBatch spriteBatch;
        private Texture2D texture;
        private Vector2 position;

        public Game1()
        {
            graphics = new GraphicsDeviceManager(this);
            Content.RootDirectory = "Content";
            position = Vector2.Zero;
        }

        /// <summary>
        /// Allows the game to perform any initialization it needs to before starting to run.
        /// This is where it can query for any required services and load any non-graphic
        /// related content.  Calling base.Initialize will enumerate through any components
        /// and initialize them as well.
        /// </summary>
        protected override void Initialize()
        {
            graphics.PreferMultiSampling = true;
            graphics.PreferredBackBufferFormat = SurfaceFormat.Color;
            graphics.PreferredDepthStencilFormat = DepthFormat.Depth16;
            graphics.SupportedOrientations = DisplayOrientation.Default;
            graphics.SynchronizeWithVerticalRetrace = true;
            graphics.ApplyChanges();
            IsFixedTimeStep = true;
            IsMouseVisible = false;

            base.Initialize();
        }

        /// <summary>
        /// LoadContent will be called once per game and is the place to load
        /// all of your content.
        /// </summary>
        protected override void LoadContent()
        {
            // Create a new SpriteBatch, which can be used to draw textures.
            spriteBatch = new SpriteBatch(GraphicsDevice);

            texture = Content.Load<Texture2D>("Texture");
        }

        /// <summary>
        /// UnloadContent will be called once per game and is the place to unload
        /// game-specific content.
        /// </summary>
        protected override void UnloadContent()
        {
            texture?.Dispose();
        }

        /// <summary>
        /// Allows the game to run logic such as updating the world,
        /// checking for collisions, gathering input, and playing audio.
        /// </summary>
        /// <param name="gameTime">Provides a snapshot of timing values.</param>
        protected override void Update(GameTime gameTime)
        {
            var state = Mouse.GetState();

            position = new Vector2(state.Position.X, state.Position.Y);

            base.Update(gameTime);
        }

        /// <summary>
        /// This is called when the game should draw itself.
        /// </summary>
        /// <param name="gameTime">Provides a snapshot of timing values.</param>
        protected override void Draw(GameTime gameTime)
        {
            GraphicsDevice.Clear(Color.CornflowerBlue);

            spriteBatch.Begin(SpriteSortMode.Deferred);
            spriteBatch.Draw(texture, position, null, Color.White, 0.0f, Vector2.Zero, 0.5f, SpriteEffects.None, 0.0f);
            spriteBatch.End();

            base.Draw(gameTime);
        }
    }
}
```

**Comment by lozzajp** [comment]
Just tested it on one machine (can test on another later).

1. MG Dev 3.6.0.1060
2. Used same square44 texture (and tried another of my own too), still called it Texture.png. (pipeline tool looked ok this time around).
3. Copy pasted above code.
4. Game ran as expected, cornflower blue and the texture followed the mouse.

@GrafSeismo are you still having troubles?

**Comment by lozzajp** [comment]
Just tried again, same conditions.

Nothing rendered first time around, I opened GamePage.xaml ran again and it worked (I recall doing this first test time).

Will test a few more project see if this is consistent.

**Comment by GrafSeismo** [comment]
With current version 3.6.0.1060 it work for me now. Thank you!
